jina/types/document/__init__.py

```diff
@@ -226,96 +229,110 @@ def __init__(
                 assert d.tags['hello'] == 'world'  # true
                 assert d.tags['good'] == 'bye'  # true
         """
-        self._pb_body = jina_pb2.DocumentProto()
-        try:
-            if isinstance(document, jina_pb2.DocumentProto):
-                if copy:
-                    self._pb_body.CopyFrom(document)
-                else:
-                    self._pb_body = document
-            elif isinstance(document, (dict, str)):
-                if isinstance(document, str):
-                    document = json.loads(document)
-
-                def _update_doc(d: Dict):
-                    for key in _all_doc_array_keys:
-                        if key in d:
-                            value = d[key]
-                            if isinstance(value, list):
-                                d[key] = NdArray(np.array(d[key])).dict()
-                        if 'chunks' in d:
-                            for chunk in d['chunks']:
-                                _update_doc(chunk)
-                        if 'matches' in d:
-                            for match in d['matches']:
-                                _update_doc(match)
-
-                _update_doc(document)
-
-                if field_resolver:
-                    document = {
-                        field_resolver.get(k, k): v for k, v in document.items()
-                    }
-
-                user_fields = set(document)
-                support_fields = set(
-                    self.attributes(
-                        include_proto_fields_camelcase=True, include_properties=False
-                    )
+        Document.CNT += 1
+        if pb_only:
+            if not isinstance(document, jina_pb2.DocumentProto) or document is None:
+                raise ValueError(
```

Comment:
can you restructure you `if` here so that the rest of the code remains untouched. Right now I see a lot of lines of changes but it seems that they only differs in ident

Reply:
s3_s3a

Reply:
I refactored the constructor